src/solver/conv_hip_implicit_gemm_fwd_xdlops.cpp

```diff
@@ -227,6 +226,40 @@ void RunCKSolution(const Handle& handle,
     }
 }
 
+InvokerFactory MakeInvokerFactoryHipImplGemmFwdXdlops(const ProblemDescription& problem, 
+    const PerformanceConfigHipImplicitGemmFwdXdlops& config) {
+
+  auto ck_args = CKArgs{problem};
+  miopenDataType_t data_type = problem.conv_problem.GetInDataType();
+  auto config_idx = config.index;
+
+
+  InvokerFactory ret = [ck_args, data_type, config_idx] (const std::vector<Kernel>& kernels) {
+        std::ignore = kernels;
+        return [ck_args, data_type, config_idx] (const Handle& handle, const AnyInvokeParams& primitive_parameters) {
+            switch(data_type)
+            {
+            case miopenInt8:
+                RunCKSolution<int8_t>(handle, primitive_parameters, ck_args, config_idx);
+                break;
+            case miopenHalf:
+                RunCKSolution<ck::half_t>(handle, primitive_parameters, ck_args, config_idx);
+                break;
+            case miopenFloat:
+                RunCKSolution<float>(handle, primitive_parameters, ck_args, config_idx);
+                break;
+            case miopenInt32:
+            case miopenInt8x4:
+            case miopenBFloat16:
+            case miopenDouble: break;
```

Comment:
This should be `MIOPEN_THROW("some description like unsopported data type")` as it is unlikely that noop is expected behaviour. I understand that invoker is not expected to be created in such case in the first place, but it's better to future-proof it with a transparent error message and option to select another solver as recovery during find stage.

Reply:
Also, it's better to move the switch outside of the invoker as Artem has suggested. That would allow to enter recovery mode on throw rather than fail the find entirely.

Reply:
After discussion with @JehandadKhan , I concluded that we shouldn't throw an exception here as the `IsApplicable` method should implement the required checks.

Reply:
@amberhassaan If you 100% sure that IsApplicable implements the necessary checks (**_and will remain valid in the future_**), then MIOPEN_THROW is just a dead code. Otherwise, please add it (or put assert at least). Pls decide yourself. [Resolved]